whylog/log_reader/read_utils.py

```diff
@@ -59,3 +70,40 @@ def get_line_containing_offset(cls, fd, offset, buf_size):
         if there is '\n' on specified offset, the previous line is returned
         """
         return cls._read_entire_line(fd, offset, buf_size)
+
+    @classmethod
+    def binary_search_left(cls, fd, left, right, value, super_parser):
```

Comment:
I doubt this works - but it will be tested in future PR, so I guess it is ok for now.